### PR TITLE
Javalab: prevent comments on projects that have not been edited

### PR DESCRIPTION
**Pull request by bencodeorg**
If a teacher views a student project that has not been edited (ie, is only starter code), we do not have a `channelId` for that project. This PR adds a message saying that review can only happen once a student has edited their code, which prevents errors that we saw prior to the bug bash where we tried check the reviewable status of a project without a channel ID, eg:

https://app.honeybadger.io/projects/3240/faults/80779436

## Testing story

Tested manually.

### CODE REVIEW

File: apps/src/templates/instructions/ReviewTab.jsx [hunk 1]
```diff
@@ -34,31 +36,35 @@ class ReviewTab extends Component {
...
 
-    codeReviewDataApi
-      .getCodeReviewCommentsForProject(channelId)
-      .done((data, _, request) => {
-        this.setState({
-          comments: data,
-          token: request.getResponseHeader('csrf-token')
+    // If there's no channelId (happens when a teacher is viewing as a student who has not done any work on a level),
+    // do not make API calls that require a channelId
+    if (channelId) {
```
Comment (sanchitmalhotra126) [inline apps/src/templates/instructions/ReviewTab.jsx]: nit: invert and early return if we don't have a channel ID?

File: apps/src/templates/instructions/ReviewTab.jsx [hunk 2]
```diff
@@ -268,18 +274,33 @@ class ReviewTab extends Component {
...
-    return (
-      <div style={styles.reviewsContainer}>
-        <div style={styles.reviewHeader}>
-          {this.renderReadyForReviewCheckbox()}
+    // channelId is not available on projects where the student has not edited the starter code.
+    // comments cannot be made on projects in this tate.
+    const projectOwnerHasEditedCode = !!getStore().getState().pageConstants
+      .channelId;
+
+    if (projectOwnerHasEditedCode) {
```
Comment (sanchitmalhotra126) [inline apps/src/templates/instructions/ReviewTab.jsx]: nit: same here, invert and early return the disabled message if `!projectOwnerHasEditedCode`?

**Review by sanchitmalhotra126 — APPROVED** [review]
Small nit but otherwise LGTM!

File: apps/src/templates/instructions/ReviewTab.jsx [hunk 3]
```diff
@@ -353,37 +359,50 @@ class ReviewTab extends Component {
...
-    return (
-      <div style={styles.reviewsContainer}>
-        <div style={styles.reviewHeader}>
-          {viewAs !== ViewType.Teacher &&
-            !errorLoadingReviewblePeers &&
-            (viewAsCodeReviewer
-              ? this.renderBackToMyProject(this.onClickBackToProject)
-              : this.renderPeerDropdown(reviewablePeers, this.onSelectPeer))}
-          {this.renderReadyForReviewCheckbox()}
+    // channelId is not available on projects where the student has not edited the starter code.
```
Comment (bencodeorg) [inline apps/src/templates/instructions/ReviewTab.jsx:362]: Git is picking this up as a diff, but it just puts the previous return statement inside of an else block.

File: apps/src/templates/instructions/ReviewTab.jsx [hunk 4]
```diff
@@ -336,30 +361,47 @@ class ReviewTab extends Component {
...
       .channelId;
 
-    if (projectOwnerHasEditedCode) {
+    if (projectOwnerHasNotEditedCode) {
+      return (
+        <div style={{...styles.reviewsContainer, ...styles.messageText}}>
+          {javalabMsg.noCodeReviewUntilStudentEditsCode()}
+        </div>
+      );
+    } else {
```
Comment (sanchitmalhotra126) [inline apps/src/templates/instructions/ReviewTab.jsx:373]: nit: don't need to nest this inside of the `else` since we're returning above

Comment (bencodeorg) [reply apps/src/templates/instructions/ReviewTab.jsx:373]: Seemed clearer to me to put it in else, but 🤷

Comment (sanchitmalhotra126) [reply apps/src/templates/instructions/ReviewTab.jsx:373]: Yeah either works! It's just one less level of nesting but I'm totally fine either way

Comment (bencodeorg) [reply apps/src/templates/instructions/ReviewTab.jsx:373]: I clearly was torn and did it one way one time, and one way the other 🤦

**Pull request merged**